### DISCUSSION TITLE
dataset scribes & people pages suggestions

### DISCUSSION THREAD

**Discussion by abubelinha** [bionomia/bionomia#350] [Ideas] [answered]
I guess this is the page to see who helped to attribute collectors and identifiers of specimens from a certain collection:
`https://bionomia.net/dataset/{datasetID}/scribes`

I selected a small example dataset ([3358 specimens,](https://www.gbif.org/dataset/e5e89b4d-bb7c-4dae-b1b4-924177e95c2b) 78 collector/identifier agents, and only 2 scribes) to illustrate a problem which is not so obvious when you look at other datasets with many more specimens and people involved:

https://bionomia.net/dataset/e5e89b4d-bb7c-4dae-b1b4-924177e95c2b/scribes

When I look at the _**scribes**_ tab I'd expect to see how much the scribes contributed to THIS dataset.
But I see two people which attributed millions of specimens to thousands of people.
Obviously that information is related to their whole contribution to Bionomia (BTW ... WOW!!).

So, there is no clue about how much they contributed to this particular dataset (they may have attributed just a few specimens or the whole collection).    I'd suggest to:
- Show prominent numbers limited to the dataset we are looking at.
- Sort the scribes by those numbers (top attributions first).
- I think the whole Bionomia project numbers can still be shown, but it would be better to see them in a smaller font size

i.e. something like this (I don't know the real numbers)

> #### 2,959 specimens attributed & 60 people helped from this dataset
> (6,523,959 specimens & 4,067 people in total)

_________________________

A similar problem occurs in the same dataset _**people**_ tab:

https://en.bionomia.net/dataset/e5e89b4d-bb7c-4dae-b1b4-924177e95c2b

In this case, each person's numbers seems to be again the total of specimens claimed across all Bionomia datasets (see [Samuel Hubbard Scudder](https://en.bionomia.net/Q723149) appears with 3860, whereas this particular dataset has only 3358 specimens).

Again, I'd rather show here numbers limited to specimens claimed from this dataset, sort people desc according to those numbers, and make their total claimed specimens' number less prominent in this page.

I think many dataset curators would find that information more relevant.

Thanks!
@abubelinha

**Comment by dshorthouse** [comment]
The above once behaved in the way you've described it, but it was simplified to the present display because it was computationally too expensive. The alternative way to do this would be to reserve the user cards (they're cached as html fragments) with their present totals for the the Profiles pages and have a stripped-down list with no counts at all elsewhere.

**Reply by abubelinha** [reply]
I see.

Then I suggest to add some text to cards so it's plain clear those are global numbers not related to any particular dataset.



When you say expensive you mean to keep cards permanently updated?

Maybe you can also do nightly/weekly/whatever updates just for those person-dataset-related numbers (i.e. keep them in text files named by personId / datasetKey) and inject them in global cards when rendering dataset-related html pages with an informative text "X as of yyyy.mm.dd"?

 (so you don't generate these X numbers on the fly)

**Reply by dshorthouse** [reply]
> When you say expensive you mean to keep cards permanently updated?

Partly, yes. These global numbers are refreshed as needed on a semi-regular basis, dependent on activity. One of these activities is bi-monthly refresh from GBIF. The underlying ~200M occurrence data are wholesale refreshed from GBIF downloads every two weeks. Any one dataset from GBIF might experience substantial flux in numbers of occurrence records, which cascades to the underlying links made to collectors/determiners in Bionomia that sometimes need repair. _Global_ attribution counts are already challenging to keep up-to-date; additionally doing so across a matrix of ~15k people (and counting) and 24k datasets (and counting) would be computationally challenging to do even on a schedule with a cache.

**Comment by dshorthouse** [comment, marked as answer]
> Then I suggest to add some text to cards so it's plain clear those are global numbers not related to any particular dataset.

That's a great idea, https://github.com/bionomia/bionomia/issues/351
